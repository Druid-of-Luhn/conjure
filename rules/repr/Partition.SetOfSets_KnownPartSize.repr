
~~> SetOfSets_KnownPartSize
~~> set of set (size n) of tau

*** partition (partSize n) from tau
    ~~> ( forAll i,j : refn , i != j . |i intersect j| = 0 )
    /\  ( forAll i : tau . exists s : refn . i in s )
    /\  ( forAll i,j : refn . |i| = |j| )
