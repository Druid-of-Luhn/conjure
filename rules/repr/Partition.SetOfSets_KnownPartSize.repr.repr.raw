RuleRepr
  { reprFilename = "rules/repr/Partition.SetOfSets_KnownPartSize.repr.repr"
  , reprName = "SetOfSets_KnownPartSize"
  , reprTemplate = DomainSet
                     { size = Nothing
                     , minSize = Nothing
                     , maxSize = Nothing
                     , attrDontCare = False
                     , element = DomainSet
                                   { size = Just ( Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_n" )
                                   , minSize = Nothing
                                   , maxSize = Nothing
                                   , attrDontCare = False
                                   , element = Identifier "tau"
                                   , representation = Nothing
                                   }
                     , representation = Nothing
                     }
  , reprPrologueStructural = Nothing
  , reprPrologueWheres = []
  , reprPrologueBindings = []
  , reprCases = [ RuleReprCase
                    { reprCasePattern = DomainPartition
                                          { element = Identifier "tau"
                                          , isRegular = False
                                          , isComplete = False
                                          , size = Nothing
                                          , minSize = Nothing
                                          , maxSize = Nothing
                                          , partSize = Just ( Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_n" )
                                          , minPartSize = Nothing
                                          , maxPartSize = Nothing
                                          , numParts = Nothing
                                          , minNumParts = Nothing
                                          , maxNumParts = Nothing
                                          , attrDontCare = False
                                          , representation = Nothing
                                          }
                    , reprCaseStructural = Just (
                                             GenericNode
                                               And
                                               [ GenericNode
                                                   And
                                                   [ ExprQuantifier
                                                       { quanName = Identifier "forall"
                                                       , quanVar = Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_i"
                                                       , quanOver = Identifier "refn"
                                                       , quanGuard = Nothing
                                                       , quanBody = ExprQuantifier
                                                                      { quanName = Identifier "forall"
                                                                      , quanVar = Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_j"
                                                                      , quanOver = Identifier "refn"
                                                                      , quanGuard = Just (
                                                                                      GenericNode
                                                                                        Neq
                                                                                        [ Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_i"
                                                                                        , Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_j"
                                                                                        ] )
                                                                      , quanBody = GenericNode
                                                                                     Eq
                                                                                     [ GenericNode
                                                                                         Card
                                                                                         [ GenericNode
                                                                                             Intersect
                                                                                             [ Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_i"
                                                                                             , Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_j"
                                                                                             ]
                                                                                         ]
                                                                                     , ValueInteger 0
                                                                                     ]
                                                                      }
                                                       }
                                                   , ExprQuantifier
                                                       { quanName = Identifier "forall"
                                                       , quanVar = Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_i"
                                                       , quanOver = Identifier "tau"
                                                       , quanGuard = Nothing
                                                       , quanBody = ExprQuantifier
                                                                      { quanName = Identifier "exists"
                                                                      , quanVar = Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_s"
                                                                      , quanOver = Identifier "refn"
                                                                      , quanGuard = Nothing
                                                                      , quanBody = GenericNode
                                                                                     Elem
                                                                                     [ Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_i"
                                                                                     , Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_s"
                                                                                     ]
                                                                      }
                                                       }
                                                   ]
                                               , ExprQuantifier
                                                   { quanName = Identifier "forall"
                                                   , quanVar = Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_i"
                                                   , quanOver = Identifier "refn"
                                                   , quanGuard = Nothing
                                                   , quanBody = ExprQuantifier
                                                                  { quanName = Identifier "forall"
                                                                  , quanVar = Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_j"
                                                                  , quanOver = Identifier "refn"
                                                                  , quanGuard = Nothing
                                                                  , quanBody = GenericNode
                                                                                 Eq
                                                                                 [ GenericNode
                                                                                     Card
                                                                                     [ Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_i"
                                                                                     ]
                                                                                 , GenericNode
                                                                                     Card
                                                                                     [ Identifier "__rules_repr_Partition_SetOfSets_KnownPartSize_repr_repr_j"
                                                                                     ]
                                                                                 ]
                                                                  }
                                                   }
                                               ] )
                    , reprCaseWheres = []
                    , reprCaseBindings = []
                    }
                ]
  }