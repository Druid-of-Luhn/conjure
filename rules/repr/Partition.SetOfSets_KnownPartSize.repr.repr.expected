
~~> SetOfSets_KnownPartSize
~~> set of set (size n) of tau

*** partition (partSize n) from tau
    ~~> (forall i : refn . forall j : refn , i != j . card(i intersect j) = 0)
        /\
        (forall i : tau . exists s : refn . i elem s)
        /\
        (forall i : refn . forall j : refn . card(i) = card(j))
